backend-app/controllers/userController.js

```diff
@@ -18,6 +29,40 @@ exports.deleteMe = async (req, res, next) => {
     }
 };
 
+
+exports.updateMe = async (req, res, next) => {
+    try {
+        // 1) Create error if user POSTs password data
+        if (req.body.password || req.body.passwordConfirm) {
+            return next(new AppError(400, 'fail', 'This route is not for password updates. Please use /updateMyPassword'), req, res, next);
+        }
+        // create error if user tries to update role
+        if (req.body.roles) {
+            return next(new AppError(400, 'fail', 'This route is not for role updates. Please use /updateRole'), req, res, next);
+        }
+        // 2) Filtered out unwanted fields names that are not allowed to be updated
+        const filteredBody = Object.keys(req.body).filter(el => el !== 'name' && el !== 'email');
```

Comment:
I think you should allow user to change his name

Reply:
sure i ll fix it in next pr